src/db/source/user.db.datasource.ts

```diff
@@ -1,12 +1,12 @@
-import { dataORM } from '@db/dbconfig';
 import { CreateUserInputModel } from '@domain/model';
 import { User } from '@entities';
 import { Service } from 'typedi';
 import { Repository } from 'typeorm';
+import { Database } from '../dbconfig';
```

Comment:
da pra usar o caminho absoluto ne

```suggestion
import { Database } from '@data/db/dbconfig';
```

Reply:
Esse caminho não existe, daí eu chamei de @db o db/index
Porém acabei de verificar que tenho acertar colocando uma pasta com nome "data" abaixo da src.
Mas se não houve problemas de dependências cíclicas faço isso que falou, se não terei que deixar assim mesmo.